File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int pointerS = 0, pointerT = 0;

        if (s.size() == 0) {
            return true;
        }

        while (pointerT < t.size()) {
            if (s[pointerS] == t[pointerT]) {
                pointerS++;
                if (pointerS == s.size()) {
                    return true;
                }
            }
            pointerT++;
        }

        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        sPtr = 0
        tPtr = 0

        if len(s) == 0:
            return True

        while tPtr < len(t) and sPtr < len(s):
            if s[sPtr] == t[tPtr]:
                sPtr += 1
                tPtr += 1
                continue
            else:
                tPtr += 1
        
        if sPtr != len(s):
            return False
        else:
            return True